pkg/plugins/builtin/httputil/httputil.go

```
@@ -110,3 +113,42 @@ func defaultHTTPClientFactory(cfg HTTPClientConfig) HTTPClient {
 	}
 	return c
 }
+
+func GetTransport(opts ...func(*http.Transport)) http.RoundTripper {
+	tr := http.DefaultTransport.(*http.Transport).Clone()
+	for _, o := range opts {
+		o(tr)
+	}
+
+	_ = http2.ConfigureTransport(tr)
+	return tr
+}
+
+func WithTLSInsecureSkipVerify(v bool) func(*http.Transport) {
+	return func(t *http.Transport) {
+		if t.TLSClientConfig == nil {
+			t.TLSClientConfig = &tls.Config{}
+		}
+		t.TLSClientConfig.InsecureSkipVerify = v
+	}
+}
+
+func WithTLSClientAuth(cert tls.Certificate) func(*http.Transport) {
+	return func(t *http.Transport) {
+		if t.TLSClientConfig == nil {
+			t.TLSClientConfig = &tls.Config{}
+		}
+		t.TLSClientConfig.Certificates = []tls.Certificate{cert}
+	}
+}
+
+func WithTLSRootCA(caCert []byte) func(*http.Transport) {
+	return func(t *http.Transport) {
+		if t.TLSClientConfig == nil {
+			t.TLSClientConfig = &tls.Config{}
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
```

it seems to fail silently ?

---

no, I will add error handling